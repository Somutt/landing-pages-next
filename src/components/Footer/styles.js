import styled, { css } from 'styled-components';
import { Wrapper as TextComponent } from '../TextComponent/styles';
import { Wrapper as SectionContainer } from '../SectionContainer/styles';

export const Wrapper = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    ${TextComponent} {
      font-size: ${theme.font.sizes.xsmall};
    }

    ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
