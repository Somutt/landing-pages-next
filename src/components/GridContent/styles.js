import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `};
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.big} 0;
  `};
`;
