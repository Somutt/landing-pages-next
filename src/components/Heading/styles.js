import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
`;

export const Title = styled.h1`
  ${({ theme, dark, size, uppercase }) => css`
    color: ${dark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
