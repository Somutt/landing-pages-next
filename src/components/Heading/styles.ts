import styled, { css } from 'styled-components';
import { DefaultTheme } from 'styled-components/dist/types';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
};

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, dark, size, uppercase }) => css`
    color: ${dark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
