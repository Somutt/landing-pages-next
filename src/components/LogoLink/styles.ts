import styled, { css } from 'styled-components';

export const Wrapper = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
      height: 2rem;
    }
  `};
`;
