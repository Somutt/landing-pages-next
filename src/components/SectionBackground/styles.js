import styled, { css } from 'styled-components';

const activateBackground = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Wrapper = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && activateBackground(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
