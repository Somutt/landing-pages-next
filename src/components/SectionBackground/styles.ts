import styled, { css } from 'styled-components';
import { DefaultTheme } from 'styled-components/dist/types';
import { SectionBackgroundProps } from '.';

const activateBackground = (theme: DefaultTheme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Wrapper = styled.div<SectionBackgroundProps>`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && activateBackground(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
