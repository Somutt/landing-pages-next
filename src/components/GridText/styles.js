import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.big};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.big};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.large};
      top: -1.5rem;
      left: -4rem;
      transform: rotate(10deg);
    }
  `}
`;
