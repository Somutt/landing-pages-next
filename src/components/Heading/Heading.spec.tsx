import { screen } from '@testing-library/react';
import { Heading } from '.';
import { theme } from '../../styles/theme';
import { renderTheme } from '../../utils/renderTheme';

describe('<Heading />', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.font.sizes.big,
      'text-transform': 'none',
    });
  });

  it('should render with white color', () => {
    renderTheme(<Heading dark={false}>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render with small heading size', () => {
    renderTheme(<Heading size='small'>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.small,
    });
  });

  it('should render with medium heading size', () => {
    renderTheme(<Heading size='medium'>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.medium,
    });
  });

  it('should render with big heading size', () => {
    renderTheme(<Heading size='big'>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.big,
    });
  });

  it('should render with large heading size', () => {
    renderTheme(<Heading size='large'>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.large,
    });
  });

  it('should render correct font size when screen is less than 768px', () => {
    renderTheme(<Heading size='large'>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyleRule('font-size', theme.font.sizes.medium, {
      media: theme.media.lteMedium,
    });
  });

  it('should render with uppercase letters', () => {
    renderTheme(<Heading uppercase>text</Heading>);
    const heading = screen.getByRole('heading', { name: /text/i });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render the heading as h6', () => {
    const { container } = renderTheme(<Heading as='h6'>text</Heading>);
    const h6 = container.querySelector('h6');

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });

  it('should render with uppercase letters', () => {
    const { container } = renderTheme(<Heading uppercase>text</Heading>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
