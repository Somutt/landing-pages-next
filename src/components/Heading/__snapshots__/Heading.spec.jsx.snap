// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render with uppercase letters 1`] = `
.c0 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  text
</h1>
`;
